sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiWriteUnshardedRecords.java

```diff
@@ -430,8 +430,13 @@ SchemaAndDescriptor getCurrentTableSchema(String stream, @Nullable TableSchema u
                 WriteStream writeStream =
                     Preconditions.checkStateNotNull(maybeDatasetService).getWriteStream(streamName);
                 if (writeStream != null && writeStream.hasTableSchema()) {
-                  currentSchema.set(writeStream.getTableSchema());
+                  TableSchema updatedFromStream = writeStream.getTableSchema();
+                  currentSchema.set(updatedFromStream);
                   updated.set(true);
+                  LOG.debug(
+                      "Updated currentSchema according to stream {}: {}",
+                      streamName,
+                      updatedFromStream);
```

Comment:
"Fetched updated schema for table {}: \n\t{}" ?

Reply:
Also this is in the context of a `DestinationState`, so we can display the table (similar to what you do in StorageApiWritesShardedRecords) using `tableUrn`.

Reply:
sounds good, added